app.py

```diff
@@ -74,31 +123,34 @@ def asynchronous_push(f_stop):
     state_display = gr.Markdown(f"Your messages: 0/{TOTAL_CNT}")
 
     # Generate model prediction
-    # Default model: distilbert-base-uncased-finetuned-sst-2-english
     def _predict(txt, state):
-        conversation_1 = Conversation(past_user_inputs=state["past_user_inputs"].copy(), generated_responses=state["generated_responses"].copy())
-        conversation_2 = Conversation(past_user_inputs=state["past_user_inputs"].copy(), generated_responses=state["generated_responses"].copy())
-        conversation_1.add_user_input(txt)
-        conversation_2.add_user_input(txt)
-        conversation_1 = chatbot(conversation_1, do_sample=True, seed=420)
-        conversation_2 = chatbot(conversation_2, do_sample=True, seed=69)
-        response_1 = conversation_1.generated_responses[-1]
-        response_2 = conversation_2.generated_responses[-1]
+        # TODO: parallelize this!
+        response_1 = chatbot_1.predict(input=txt)
+        response_2 = chatbot_2.predict(input=txt)
+        response_3 = chatbot_3.predict(input=txt)
+        response_4 = chatbot_4.predict(input=txt)
+
+        response2model = {}
```

Comment:
This is how I link response <--> model so we can track which model was preferred. Open to suggestions if you think there's a better way!